### PR TITLE
Fully support min distance to contact during planning

### PR DESCRIPTION
**Pull request by marip8**
Currently the contact check task checks if a trajectory is a specified minimum distance from contact, but none of the planners create trajectories considering this constraint. This PR adds this minimum distance to contact value to the collision margin config of the custom Descartes, OMPL, and Trajopt profiles

Builds on #87

### CODE REVIEW

**Review by marrts — APPROVED** [review]

**Comment by marip8** [comment]
This seems to fail `Descartes` often for non-zero values of `min_contact_dist` for our applications because the TCP link is generally very close to the scan link, but not generally added to the list of touch links. We may need to consider adding an allowed collision between a tool link and the scan link specifically in the Descartes (and possibly TrajOpt) collision checking config to account for this

**Comment by marrts** [comment]
Instead of adding an allowed collision you could add a unique collision pair between the tcp link and the scan that is 0. Would be a nice addition to this demo code as well.

File: snp_motion_planning/src/planning_server.cpp [hunk 1]
```diff
@@ -369,10 +381,17 @@ class PlanningServer
...
+        auto contact_check_profile =
+            std::make_shared<tesseract_planning::ContactCheckProfile>(contact_check_lvs, min_contact_dist);
+        contact_check_profile->config.contact_request.type = tesseract_collision::ContactTestType::FIRST;
+        contact_check_profile->config.contact_manager_config.margin_data_override_type =
+            tesseract_common::CollisionMarginOverrideType::MODIFY;
+        for (const UniqueCollisionPair& pair : collision_pairs)
+          contact_check_profile->config.contact_manager_config.margin_data.setPairCollisionMargin(
+              pair.first, pair.second, pair.distance);
+
+        profile_dict->addProfile<tesseract_planning::ContactCheckProfile>(CONTACT_CHECK_DEFAULT_NAMESPACE, PROFILE,
```
Comment (marrts) [inline snp_motion_planning/src/planning_server.cpp:393]: I feel like creating the `ContactCheckProfile` should be a method in the `planner_profiles.hpp` file.

File: snp_motion_planning/launch/planning_server.launch.xml [hunk 2]
```diff
@@ -19,7 +20,7 @@
   <arg name="max_translational_acc" default="0.100"/>
   <arg name="max_rotational_acc" default="3.14"/>
   <arg name="check_joint_accelerations" default="false"/>
-  <arg name="contact_check_distance" default="0.0"/>
+  <arg name="min_contact_distance" default="0.0"/>
```
Comment (marrts) [inline snp_motion_planning/launch/planning_server.launch.xml:23]: I know this is what we had before, but a default contact distance of 0.0 seems too close to me.

File: snp_motion_planning/src/planning_server.cpp [hunk 3]
```diff
@@ -339,21 +341,31 @@ class PlanningServer
 
       tesseract_planning::ProfileDictionary::Ptr profile_dict =
           std::make_shared<tesseract_planning::ProfileDictionary>();
+
       // Add custom profiles
       {
+        auto min_contact_dist = get<double>(node_, MIN_CONTACT_DIST_PARAM);
+
+        std::vector<UniqueCollisionPair> collision_pairs;
```
Comment (marrts) [inline snp_motion_planning/src/planning_server.cpp:349]: Add a comment to explain a little what is happening

File: snp_motion_planning/src/planner_profiles.hpp [hunk 4]
```diff
@@ -16,8 +16,22 @@ static const std::string MIN_LENGTH_DEFAULT_NAMESPACE = "MinLengthTask";
 static const std::string CONTACT_CHECK_DEFAULT_NAMESPACE = "DiscreteContactCheckTask";
 static const std::string ISP_DEFAULT_NAMESPACE = "IterativeSplineParameterizationTask";
 
+struct UniqueCollisionPair
```
Comment (marrts) [inline snp_motion_planning/src/planner_profiles.hpp:19]: Add a comment to explain what this is

File: snp_motion_planning/launch/planning_server.launch.xml [hunk 5]
```diff
@@ -4,7 +4,8 @@
   <arg name="robot_description"/>
   <arg name="robot_description_semantic"/>
   <arg name="verbose" default="False"/>
-  <arg name="touch_links" default="[]"/>
+  <arg name="scan_disabled_contact_links" default="[]"/>
+  <arg name="scan_reduced_contact_links" default="[]"/>
```
Comment (marrts) [inline snp_motion_planning/launch/planning_server.launch.xml:8]: Add descriptions so we don't just rely on the name of the argument

File: snp_motion_planning/src/planning_server.cpp (hunk 1 above)
Comment (marip8) [reply snp_motion_planning/src/planning_server.cpp:393]: I thought about that too; I'll move it

**Pull request merged**